dotnet/src/SemanticKernel.Abstractions/Filters/AutoFunctionInvocation/IAutoFunctionInvocationFilter.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel;
+
+#pragma warning disable CA1716 // Identifiers should not match keywords (Func<FunctionCallInvocationContext, Task> next)
```

Comment:
Not sure why are we moving `Calling` away in favor of `Invocation` as this clashes with the already know concept `KernelFunctions.InvokeAsync`.

```suggestion
#pragma warning disable CA1716 // Identifiers should not match keywords (Func<AutoFunctionInvocationContext, Task> next)
```

Reply:
Based on our discussion with @matthewbolanos , there are 3 phases: Call -> Invoke -> Result. `Call` in this case means request to call something, but it's just a request, without actual action. `Invoke` it's when we are performing some action (invoking a function), and this is exactly when filter will be triggered, so that's why filter and context model naming was updated.